### PR TITLE
test: added unit tests and test assets

### PR DESCRIPTION
**Pull request by masterchief164**

### CODE REVIEW

**Comment by adamjonas** [comment]
let's add some instructions on running the test suite on the readme

File: test/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        playlist = application.find_source_type("PLPQwGV1aLnTuN6kdNWlElfr2tzigB9Nnj") == "playlist"
+
+        audio = application.find_source_type(
+            "https://anchor.fm/s/12fe0620/podcast/play/32260353/https%3A%2F%2Fd3ctxlq1ktw2nl.cloudfront.net%2Fstaging"
+            "%2F2021-3-26%2Fdc6f12e7-a547-d872-6ef6-7acfe755a692.mp3") == "audio"
+        self.assertTrue(audio and playlist and video)
+
+    def test_download_audio_file(self):
+        audio = application.get_audio_file(
+            "https://anchor.fm/s/12fe0620/podcast/play/32260353/https%3A%2F%2Fd3ctxlq1ktw2nl.cloudfront.net%2Fstaging"
```
Comment (adamjonas) [inline test/test_cli.py:28]: A metric of usefulness for this test suite should be speed. If you are going to test this (which I think is debatable) you should fetch the shortest file possible. On poor internet this took me ~25 seconds. That's way too much.

In general though, I don't think it's very useful to test the fetching of youtube videos or audio files. Maybe that can be a sanity check test but it shouldn't be run often.

File: test/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    def test_download_video_file(self):
+        url = "https://www.youtube.com/watch?v=B0HW_sJ503Y"
+        video = application.download_video(url)
+        self.assertTrue(os.path.isfile(video) and os.path.isfile(video[:-3] + "description"))
+        print()
+        os.remove(video)
+        os.remove(video[:-3] + "description")
+
+    def test_chapter_creation(self):
```
Comment (adamjonas) [inline test/test_cli.py:48]: This is much more inline with what I'd suggest. Testing features based on mocks.

**Review by adamjonas — COMMENTED** [review]
The unit tests are of moderate use and it doesn't hurt for them to exist, but I would suggest putting more work into functional tests that document the features of the program and ensure that new features don't introduce regressions.

It also means that new features are accompanied by a nice test which should help you iterate faster by being able to run offline mock data through the program rather than pulling live data.

File: test/test_cli.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        with open("testAssets/test_video.chapters", "r") as file:
+            chapters = str(file.read().strip("\n"))
+            assert data == chapters
+    os.remove("testAssets/test_video_1.chapters")
+
+
+@pytest.mark.main
+def test_split_video():
+    chapters = application.read_description("testAssets/test_video.description")
+    application.split_mp4(chapters, "testAssets/test_video.mp4", "testAssets/test_video")
```
Comment (adamjonas) [inline test/test_cli.py:71]: I don't think it makes sense to have such a big asset in the project. Can we find something smaller?

Comment (masterchief164) [reply test/test_cli.py:71]: I've reduced the size of the video file to 3 mbs. I don't think smaller is possible.

Comment (masterchief164) [reply test/test_cli.py:71]: I've reduced the size of the video file to 3 mbs. I don't think smaller is possible.

Comment (adamjonas) [reply test/test_cli.py:71]: Screen Shot 2023-02-02 at 1 42 12 PM

File: Readme.md [hunk 4]
```diff
@@ -39,6 +39,24 @@ To check the version:
...
+`pytest -v -m main -s`
+
+To run the feature tests
+
+`pytest -v -m feature -s`
+
+To run the full test suite
+
+`pytest -v -s`
+
```
Comment (adamjonas) [inline Readme.md:59]: Install pytest (should be added to readme). Ran `pytest -v -s`

Screen Shot 2023-02-02 at 1 48 59 PM

File: test/test_cli.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+
+@pytest.mark.main
+def test_find_source_type():
+    video = application.check_source_type("B0HW_sJ503Y") == "video"
+
+    playlist = application.check_source_type("PLPQwGV1aLnTuN6kdNWlElfr2tzigB9Nnj") == "playlist"
+
+    audio = application.check_source_type(
+        "https://anchor.fm/s/12fe0620/podcast/play/32260353/https%3A%2F%2Fd3ctxlq1ktw2nl.cloudfront.net%2Fstaging"
```
Comment (adamjonas) [inline test/test_cli.py:24]: You should use a shorter file. 26 min isn't necessary.

File: test/test_cli.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    os.remove(audio)
+
+
+@pytest.mark.main
+def test_process_mp3():
+    result = application.process_mp3("testAssets/audio.mp3", "tiny")
+    with open("testAssets/transcript.txt", "r") as file:
+        data = str(file.read().strip("\n"))
+        result = result.strip(" ")
+        assert data == result
```
Comment (adamjonas) [inline test/test_cli.py:43]: Ran 3 times, could not get this to pass. Additionally, it takes about 95% of the time (~90 sec) to run this test. This should be pulled for now and rethought.

Screen Shot 2023-02-02 at 1 57 41 PM

File: test/test_cli.py (hunk 5 above)
Comment (masterchief164) [reply test/test_cli.py:22]: I'm in fact scrapping this test as this just checks if pywhisper is working properly or not and I don't think that it needs to be tested.

File: test/test_cli.py (hunk 6 above)
Comment (masterchief164) [reply test/test_cli.py]: I'm infact scrapping this test as I don't think that part needs to be tested.

**Pull request merged**